contrib/ros/envs/ros_env.py

```diff
@@ -1,41 +1,164 @@
+from geometry_msgs.msg import Pose, Point
 import numpy as np
-import rospy
 
 from rllab.core.serializable import Serializable
-from rllab.envs.base import Env
+from rllab.envs.base import Env, Step
 from rllab.misc.ext import get_seed
+from rllab.spaces import Box
+
+from contrib.ros.util.common import rate_limited
+from contrib.ros.util.gazebo import Gazebo
 
 
 class RosEnv(Env, Serializable):
     """
     Superclass for all ros environment
     """
 
-    def __init__(self):
+    def __init__(self, task_obj_mgr, robot, simulated=False):
```

Comment:
Assuming there are objects in a specific task makes this a very particular RosEnv. What if I'm using RosEnv for a walking robot?

Reply:
What I'm saying is, if you make RosEnv too specific then it's not a very useful base class.

What was your reasoning for combining SawyerEnv and RosEnv?

Reply:
I think RosEnv can provide generative ros env functionality and Robot interface can provide e.g. sawyer robot interface, so SawyerEnv is useless. I think RosEnv can work for a walking robot, if we can implement the walking robot interface which inherits the Robot class.